### ISSUE TITLE
"Waiting for changelog lock..." for Flowable tables

### ISSUE THREAD

**Issue by nictas** [cloudfoundry-incubator/multiapps-controller#969]
Flowable is using Liquibase to manage the state of some of its database entities. However, due to Liquibase's locking mechanism, it is sometimes possible to run into a situation where the deploy-service is unable to start (read [DATABASECHANGELOGLOCK table](https://docs.liquibase.com/concepts/basic/databasechangeloglock-table.html) first if you're unfamiliar with the aforementioned mechanism).

Steps to reproduce:
1. Start the deploy-service application;
2. Stop it immediately after Liquibase updates the FLW_EV_DATABASECHANGELOGLOCK table and "acquires the lock";
_Note: Normally, the tables Liquibase uses to store information about executed changesets are called DATABASECHANGELOG and DATABASECHANGELOGLOCK, but Flowable has its own separate version of these tables and they are prefixed with "FLW_EV\_"._
3. Subsequent attempts to start the deploy-service will timeout and the following INFO message will be seen in the logs of the application:
```
2020-10-01T12:14:50.98+0300 [APP/PROC/WEB/0] OUT { "written_at":"2020-10-01T09:14:50.982Z","written_ts":50203636370755,"tenant_id":"-","component_type":"application","component_id":"d5f2581f-84a4-4b25-82d4-1aaee747017d","space_name":"deploy-service","component_name":"deploy-service","component_instance":"0","organization_id":"117c0074-fcad-464a-a95c-76b1bc5869f0","correlation_id":"-","organization_name":"SAP_xsa","space_id":"447adb6d-6887-49be-b1aa-2347ae675dc1","container_id":"10.0.137.5","type":"log","logger":"liquibase.lockservice.StandardLockService","thread":"localhost-startStop-1","level":"INFO","categories":[],"msg":"Waiting for changelog lock...." }
```

This lock must then be manually "released" by updating the FLW_EV_DATABASECHANGELOGLOCK table. We've had this issue in the past with our own tables and we fixed it by extending the standard Liquibase locking mechanism (see [RecoveringLockService](https://github.com/cloudfoundry-incubator/multiapps-controller/blob/c29d2f4478319605ae79b5203c2e6756edfe3449/multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/liquibase/RecoveringLockService.java)).

Unfortunately, either this mechanism has stopped working at some point, or it isn't used in the Flowable-Liquibase integration at all. We should investigate and fix this.

**Comment by nictas** [comment]
Turns out that our `RecoveringLockService` is still being used for "our" Liquibase tables (`DATABASECHANGELOG` and `DATABASECHANGELOGLOCK`), but not for the Flowable tables (`FLW_EV_DATABASECHANGELOG` and `FLW_EV_DATABASECHANGELOGLOCK`). This is probably due to the way the lock service is registered in Liquibase:
https://github.com/cloudfoundry-incubator/multiapps-controller/blob/c29d2f4478319605ae79b5203c2e6756edfe3449/multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/listeners/LiquibaseServicesRegistrator.java#L24

**Comment by nictas** [comment]
Liquibase uses the Java Service Provider Interface (SPI) to detect custom lock services. Registering ours via this mechanism seems to resolve the issue (see #970).

**Comment by nictas** [comment]
Also see [LockServiceFactory](https://github.com/liquibase/liquibase/blob/bdb2f69206e8c4e08a706986022bd4351407cd3a/liquibase-core/src/main/java/liquibase/lockservice/LockServiceFactory.java#L43) and [StandardServiceLocator](https://github.com/liquibase/liquibase/blob/90cd7532ababa6db46f4f344050890b465609932/liquibase-core/src/main/java/liquibase/servicelocator/StandardServiceLocator.java#L21) from [liquibase](https://github.com/liquibase/liquibase).

**Comment by nictas** [comment]
Fixed with #970.
